lib/srv/app/server.go

```diff
@@ -828,6 +829,11 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if !ok {
 		s.log.Errorf("unable to extract connection from context")
 	}
+	if r.Body != nil {
+		// no need to worry about closing the body, [http.Server] does that
+		// for us even if the body is reassigned in a handler like this.
+		r.Body = io.NopCloser(io.LimitReader(r.Body, maxHTTPRequestBodySize))
+	}
```

Comment:
good catch - aws, azure, and gcp handlers all called `GetAndReplaceRequestBody`, so they were already using the 10MB limit. Now they would be using 70MB.

TCP apps are not affected at all obviously.

That leaves the HTTP JWT token forwarder: I don't see anywhere we buffer the entire payload for JWT apps, so I can just remove the limit for those requests.

Aside, I don't think websockets are an issue at all. Those will upgrade to ws protocol, which means calling Hijack, which does not use the request's body. It also shouldn't be used after the upgrade:
>	// After a call to Hijack, the original Request.Body must not
	// be used. The original Request's Context remains valid and
	// is not canceled until the Request's ServeHTTP method
	// returns.

Unless I've missed something, which is definitely possible since I haven't worked with web sockets much.

@smallinsky 
> if is possible to check if this is AWS access lambda request looking at URL or r.Headers and extend the limit only for AWS lambda requests ?

It will be easier to just raise the limit for all AWS requests instead of trying to detect lambda, and no less secure.

## What I've changed

I've moved the request body size limiting into each of aws, gcp, azure handlers: [c75dfd2](https://github.com/gravitational/teleport/pull/40242/commits/c75dfd27c2f407cc2f37c2d887653a6f490dd95d)

This makes the lambda handler test I added go through the path where the limiting is applied, so that's a nice side-benefit.

azure and gcp I changed to 10MB (prior behavior).
aws I changed to 70MB for any requests.

I want to keep the limiter out of  `GetAndReplaceRequestBody` because it couples every caller to that limit, which makes it hard to untangle later when one of them breaks (like aws lambda request uploads).